superset-frontend/src/explore/components/AdhocFilterOption.jsx

```diff
@@ -60,16 +60,27 @@ class AdhocFilterOption extends React.PureComponent {
     // isNew is used to indicate whether to automatically open the overlay
     // once the overlay has been opened, the metric/filter will never be
     // considered new again.
-    this.props.adhocFilter.isNew = false;
     this.setState({ overlayShown: true });
   }
 
   onOverlayExited() {
     this.setState({ overlayShown: false });
   }
 
-  closeFilterEditOverlay() {
-    this.setState({ overlayShown: false });
+  getContent() {
+    const { adhocFilter } = this.props;
+    adhocFilter.isNew = false;
+    return (
+      <AdhocFilterEditPopover
+        onResize={this.onPopoverResize}
+        adhocFilter={adhocFilter}
+        onChange={this.props.onFilterEdit}
+        onClose={this.onOverlayExited}
+        options={this.props.options}
```

Comment:
NIT: Seems like we might as well destructure these other props above, for consistency.

Reply:
Done